vespa/application.py

```diff
@@ -1096,7 +1110,19 @@ async def query(
             json=await r.json(), status_code=r.status, url=str(r.url)
         )
 
-    @retry(wait=wait_exponential(multiplier=1), stop=stop_after_attempt(3))
+    @retry(
+        wait=wait_exponential(multiplier=1),
+        retry=retry_any(
+            retry_if_exception(lambda x: True),
```

Comment:
Could consider being more selective here, later, but I don't know what errors/exceptions to expect from the HTTP library, and I couldn't find documentation of that either. We want to retry transport-related failures, but not SSL-related, and other user errors.